source/abstract/memory/Memory.hh

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <Logger.hh>
+
+#include <cstdint>
+
+// We can't include Common.hh, because we have a cyclic dependency
```

Comment:
Maybe we make a Types.hh that just has these? Or is that overkill

Reply:
Definitely overkill. This is designed to be `abstract/memory`'s `Common.hh`.